### PR TITLE
Prepend prefix for MAR operations

### PR DESCRIPTION
**Pull request by adityapatwardhan**
# PR Summary

- Prepend the standardized `psresource` MAR prefix
- Block publish to MAR
- Recognize mcr.microsoft.com as ContainerRegistry repository
- Add tests

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Documentation needed]()
        - [ ] Issue filed: 
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [ ] [Documentation needed]()
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary.

### CODE REVIEW

**Review by anamnavi — APPROVED** [review]
LGTM. Eventually maybe we add a test for find and install

File: src/code/InternalHooks.cs [hunk 1]
```diff
@@ -2,6 +2,7 @@
 // Licensed under the MIT License.
 
 using System.Reflection;
+using Azure.Core;
```
Comment (adityapatwardhan) [inline src/code/InternalHooks.cs]: Checking if this is needed

Comment (adityapatwardhan) [reply src/code/InternalHooks.cs]: Removed

**Review by alerickson — APPROVED** [review]

File: src/code/RepositorySettings.cs [hunk 2]
```diff
@@ -862,7 +862,7 @@ private static PSRepositoryInfo.APIVersion GetRepoAPIVersion(Uri repoUri)
                 // repositories with Uri Scheme "temp" may have PSPath Uri's like: "Temp:\repo" and we should consider them as local repositories.
                 return PSRepositoryInfo.APIVersion.Local;
             }
-            else if (repoUri.AbsoluteUri.EndsWith(".azurecr.io") || repoUri.AbsoluteUri.EndsWith(".azurecr.io/"))
+            else if (repoUri.AbsoluteUri.EndsWith(".azurecr.io") || repoUri.AbsoluteUri.EndsWith(".azurecr.io/") || repoUri.AbsoluteUri.Contains("mcr.microsoft.com") || repoUri.AbsoluteUri.Contains("mcr.microsoft.com/"))
```
Comment (anamnavi) [inline src/code/RepositorySettings.cs]: wouldn't `repoUri.AbsoluteUri.Contains("mcr.microsoft.com")` account for `repoUri.AbsoluteUri.Contains("mcr.microsoft.com/")`?

**Review by anamnavi — APPROVED** [review]

File: src/code/RepositorySettings.cs (hunk 2 above)
Comment (adityapatwardhan) [reply src/code/RepositorySettings.cs]: true.. I thought I used EndsWith. Updating.

Comment (adityapatwardhan) [reply src/code/RepositorySettings.cs]: Fixed

**Pull request merged**